src/config.js

```diff
@@ -82,7 +80,11 @@ const makeConfig = function (override = {}) {
  */
 const mergeRequestConfig = function (config, req) {
   const requestConfig = req.cache
-  const mergedConfig = merge({}, config, omit(requestConfig, disallowedPerRequestKeys))
+  if (requestConfig) {
+    disallowedPerRequestKeys.forEach(key => requestConfig[key] ? (delete requestConfig[key]) : undefined)
+  }
+
+  const mergedConfig = { ...config, ...requestConfig }
```

Comment:
Only found concern about not using `_.merge` is about composing the `config.cache.exclude` object per request.

I would suggest something like this:
```suggestion
  const mergedConfig = {
    ...config,
    ...requestConfig,
    exclude: {
      ...config.exclude,
      ...requestConfig.exclude
    }
  }
```

No need to apply this everywhere the config is being merged IMO.

Reply:
Makes sense. The `Should merge the permitted keys` test then fails, though. I think that's the fault of the test, though. It uses `exclude: true,` in the requestConfig. Couldn't find anything about that. `exclude.js` treats `exclude` as an Object, only. Correct?

Reply:
https://github.com/RasCarlito/axios-cache-adapter/blob/7e55d60bb6b6176003813498a8dc5bc1dc4270cb/src/config.js#L49 looks like another one.

Reply:
You're right about `exclude: true` being wrong, thanks for fixing that!

No need to change the other line mentioned above in the `makeConfig` method because it is only used when first setting up a cache instance, so it is not overriding a previously existing cache config.